### PR TITLE
[Fix] Store filtering in its original form in .elastic-connectors-sync-job

### PR DESCRIPTION
**Pull request by timgrein**
## Closes https://github.com/elastic/enterprise-search-team/issues/3950 ##

The frontend expects the advanced snippet payload to be stored under "value" in .elastic-connectors-sync-jobs to correctly show it in the sync-flyout. As ES doesn't have a dedicated array type it'll store a single filtering/filter instance as object directly. This will therefore be fixed in the FE with https://github.com/elastic/kibana/pull/151175.

#### Pre-Review Checklist
- [x] Covered the changes with automated tests
- [x] Tested the changes locally by running `make test` and `make ftest NAME=mysql`
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
~- [ ] Considered corresponding documentation changes~
~- [ ] Contributed any configuration settings changes to the configuration reference~

### CODE REVIEW

File: connectors/byoc.py [hunk 1]
```diff
@@ -309,14 +305,12 @@ def __init__(self, filter_=None):
...
-        advanced_rules = filter_.get("advanced_snippet", {})
-
-        self.advanced_rules = advanced_rules.get("value", advanced_rules)
+        self.advanced_rules = filter_.get("advanced_snippet", {})
         self.basic_rules = filter_.get("rules", [])
         self.validation = filter_.get("validation", {"state": "", "errors": []})
 
     def get_advanced_rules(self):
-        return self.advanced_rules
+        return self.advanced_rules.get("value", self.advanced_rules)
```
Comment (artem-shelkovnikov) [inline connectors/byoc.py:313]: So we return content of "advanced_snippet.value" if there's anything inside, otherwise content of "advanced_snippet"?

Comment (timgrein) [reply connectors/byoc.py:313]: It's probably better to return `{}` as default. Shouldn't happen anyway, but it's more obvious, that "value" doesn't exist.

Comment (artem-shelkovnikov) [reply connectors/byoc.py:313]: Yeah I saw original defaulting logic used `{}`, probably indeed better default to it

Comment (timgrein) [reply connectors/byoc.py]: Adjusted it 👍

**Review by artem-shelkovnikov — APPROVED** [review]

**Comment by wangch079** [comment]
Hi @timgrein , are you going to backport this to 8.7? If so, I think you should add the label `v8.7.0` instead of `v8.7.0.0`

**Comment by seanstory** [comment]
The backportrc utilizes the first two digits after the `v`, so either should work. But I bet this PR's source branch wasn't rebased onto `main` after the backportrc got changed to map `8.7` to a branch vs main, which is why the backport failed. I'd recommend manually running the backport command, Tim, and just linking the PR.

**Comment by wangch079** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.7|[](https://github.com/elastic/connectors-python/pull/540)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Comment by wangch079** [comment]
Manually generated the backport PR: #540 . Turns out some changes are already in 8.7 branch.

**Comment by wangch079** [comment]
Manually generated the backport PR: #540 . Turns out some changes are already in 8.7 branch.

**Pull request merged**